iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/EntrypointTreeImpl.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.tree.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import org.sonar.iac.common.api.tree.Tree;
+import org.sonar.iac.docker.tree.api.EntrypointTree;
+import org.sonar.iac.docker.tree.api.ExecFormLiteralTree;
+import org.sonar.iac.docker.tree.api.ExecFormTree;
+import org.sonar.iac.docker.tree.api.SeparatedList;
+import org.sonar.iac.docker.tree.api.ShellFormTree;
+import org.sonar.iac.docker.tree.api.SyntaxToken;
+
+public class EntrypointTreeImpl extends InstructionTreeImpl implements EntrypointTree {
+
+  private final ExecFormTree execForm;
+  private final ShellFormTree shellForm;
+
+  public EntrypointTreeImpl(SyntaxToken keyword, @Nullable ExecFormTree execForm, @Nullable ShellFormTree shellForm) {
```

Comment:
You can make this class much simpler. Trust the TreeFactory. Store a DockerTree of like arguments. This field does not need a getter for know. In  entrypointArguments() or maybe better arguments you can return all the arguments. Maybe add another method to be able to check which form is used. 
All in all this will reduce the complexity of the implementation, constructor and getters.